src/components/modals/PackActivitiesOffsetModal.svelte

```diff
@@ -0,0 +1,165 @@
+<script lang="ts">
+  import { createEventDispatcher } from 'svelte';
+  import { getTarget } from '../../utilities/generic';
+  import { convertDurationStringToInterval } from '../../utilities/time';
+  import { tooltip } from '../../utilities/tooltip';
+  import Modal from './Modal.svelte';
+  import ModalContent from './ModalContent.svelte';
+  import ModalFooter from './ModalFooter.svelte';
+  import ModalHeader from './ModalHeader.svelte';
+
+  const dispatch = createEventDispatcher();
+
+  let direction: 'Left' | 'Right' = 'Left';
+  let gapOffsetString: string = '0d 0h 0m 0s 0ms 0us';
+  let gapOffsetError: string | null = '';
+  let disabled: boolean = false;
+
+  function setDirection(dir: 'Left' | 'Right') {
+    direction = dir;
+  }
+
+  function cancel() {
+    dispatch('cancel');
+  }
+
+  function pack() {
+    dispatch('pack', { direction, gapOffset: gapOffsetString });
+  }
+
+  function onUpdateStartOffset(event: Event) {
+    const { value } = getTarget(event);
+    try {
+      convertDurationStringToInterval(`${value}`);
+      gapOffsetError = `${value}`.includes('-') ? 'Negative offsets are not allowed' : '';
+    } catch (error: any) {
+      gapOffsetError = error.message;
+    }
+  }
+</script>
+
+<div class="compact-modal">
+  <Modal height={150} width={300}>
+    <ModalHeader on:close={cancel}>Pack Directive(s)</ModalHeader>
+    <ModalContent>
+      <div class="form-container">
+        <div class="row">
+          <div class="label">Direction</div>
+          <div class="toggle-group">
+            <button class="toggle {direction === 'Left' ? 'active' : ''}" on:click={() => setDirection('Left')}>
+              Left
+            </button>
+            <button class="toggle {direction === 'Right' ? 'active' : ''}" on:click={() => setDirection('Right')}>
+              Right
+            </button>
+          </div>
+        </div>
+
+        <div class="row">
+          <label
+            class="label"
+            use:tooltip={{ content: 'The gap between directives when packed ', placement: 'top' }}
+            for="start-offset"
```

Comment:
For accessibility, the `for` must match an input element's `name` or `id`. The `for` should have the value `gap-offset` to match the `input` element's name or the `input` element needs to be updated.